sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1092,6 +1093,35 @@ val GPU_COREDUMP_PIPE_PATTERN = conf("spark.rapids.gpu.coreDump.pipePattern")
       .checkValue(v => v > 0, "The thread count must be greater than zero.")
       .createWithDefault(MULTITHREAD_READ_NUM_THREADS_DEFAULT)
 
+  val MULTITHREAD_READ_MEM_LIMIT = conf("spark.rapids.sql.multiThreadedRead.memoryLimit")
+      .doc("The maximum memory capacity in bytes to use for reading files in parallel. " +
+        "This can not be changed at runtime after the executor has started. And if 0, it " +
+        "will be set with 90% of `spark.executor.memoryOverhead`")
```

Comment:
spark-rapids has added host memory limits (default off) and we either get configured with `offHeap.limit` or we detect how much memory we should use for off heap. The memory limits for the scan, should follow the same pattern and use the already pre-determined limit, either via the config or automatically (https://github.com/NVIDIA/spark-rapids/blob/c0df508c3d158b9c6924ff55c02dae13a0bd6142/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala#L484). 

I am wondering if the scan limit equals the host memory we determine we have available in `GpuDeviceManager` by default. We could tune it down manually, so it would be nice to have a config for that.